### DISCUSSION TITLE
Elsa 2.0 Sample Showing Reference Last Activity Result

### DISCUSSION THREAD

**Discussion by jakenuts** [elsa-workflows/elsa-core#771] [Ideas]
I'm trying to build a photo processing pipeline and I was wondering how to reference the last result in Elsa 2.0. It would also be cool to see a sample where the "variables" being passed are poco model objects rather than primitive types. For instance, a pipeline might be "download this photo from the web, resize it, remove the background and save it to S3". I don't expect to serialize the actual photo but rather a reference to it's location and the parameters for each transformation step.

Thanks!

**Comment by sfmskywalker** [comment]
In Elsa 2, the closest thing that comes to last result is a activity's output, which automatically flows as input into the next activity.
This means there are two ways to access an activity's output:

Examples using JS:
1. `input` // This is a global variable that is set during activity execution. It contains whatever object was received as input.
2. `MyActivityName.Output` // The script engine registers expando objects using the name of the activities and storing their output values in the Output field.

The objects passed as output and received as input can be primitive types as well as complex types, which are just .NET objects.

We will be working on documentation and examples hopefully starting May.
